src/rules/cognitive-complexity.ts

```diff
@@ -339,7 +339,9 @@ const rule: TSESLint.RuleModule<string, (number | 'metric' | 'sonar-runtime')[]>
     }
 
     function visitLogicalExpression(logicalExpression: TSESTree.LogicalExpression) {
-      if (isJsxShortCircuitNode(logicalExpression)) {
+      const jsxShortCircuitNodes = getJsxShortCircuitNodes(logicalExpression);
+      if (jsxShortCircuitNodes) {
```

Comment:
`jsxShortCircuitNodes.length > 0`?

Reply:
It can also be `null` and if not its length can't be `0`, so it could be 
`jsxShortCircuitNodes != null`
so that's why the truthy check seemed sufficient. But I will change it to be clearer.